### PR TITLE
add notch filter

### PR DESCRIPTION
**Pull request by ahnitz**

### CODE REVIEW

File: pycbc/filter/resample.py [hunk 1]
```diff
@@ -178,6 +178,29 @@ def resample_to_delta_t(timeseries, delta_t, method='butterworth'):
...
                  numpy.dtype('float64'): lal.HighPassREAL8TimeSeries}
 
+def notch_fir(timeseries, f1, f2, order, beta=5.0):
+    """ notch filter the time series using an FIR filtered generated from 
+    the ideal response passed through a kaiser window (beta = 5.0)
+
+    Parameters
+    ----------
+    Time Series: TimeSeries
+        The time series to be low-passed.
```
Comment (tdent) [inline pycbc/filter/resample.py]: fix comment: notch filtered not low-passed

File: pycbc/filter/resample.py [hunk 2]
```diff
@@ -178,6 +178,29 @@ def resample_to_delta_t(timeseries, delta_t, method='butterworth'):
...
+    f1: float
+        The start of the frequency suppression.
+    f2: float
+        The end of the frequency suppression.
+    order: int
+        Number of corrupted samples on each side of the time series
```
Comment (tdent) [inline pycbc/filter/resample.py]: Preferable to suggest some values for 'order' rather than leaving it entirely up to the user?

**Comment by tdent** [comment]
also codacy didn't seem to be happy but I can't work out why .. duplications??

**Comment by spxiwh** [comment]
Codacy is unhappy with the trailing whitespace.

File: pycbc/filter/resample.py (hunk 2 above)
Comment (ahnitz) [reply pycbc/filter/resample.py]: I'm not sure about the advice to leave at the moment.

Comment (tdent) [reply pycbc/filter/resample.py]: Well, what criteria would you use to choose 'order'.  Presumably the desired 'sharpness' (frequency resolution) of the notch sides expressed as a number of samples.  So n ~ f_s/df where f_s is the sampling frequency.

File: pycbc/filter/resample.py [hunk 3]
```diff
@@ -178,6 +178,34 @@ def resample_to_delta_t(timeseries, delta_t, method='butterworth'):
 _highpass_func = {numpy.dtype('float32'): lal.HighPassREAL4TimeSeries,
                  numpy.dtype('float64'): lal.HighPassREAL8TimeSeries}
 
+def notch_fir(timeseries, f1, f2, order, beta=5.0):
+    """ notch filter the time series using an FIR filtered generated from
+    the ideal response passed through a kaiser window (beta = 5.0)
```
Comment (tdent) [inline pycbc/filter/resample.py]: prefer to say *time domain* kaiser window (if this is what the scipy function does)

File: pycbc/filter/resample.py [hunk 4]
```diff
@@ -178,6 +178,34 @@ def resample_to_delta_t(timeseries, delta_t, method='butterworth'):
...
+
+    The suppression of the notch filter is related to the bandwidth and
+    the number of samples in the filter length. For a few Hz bandwidth,
+    typically severall thousand or ten's of thousands of samples are
```
Comment (tdent) [inline pycbc/filter/resample.py]: this was more or less what I had in mind .. however, the number of samples is obviously dependent on the sampling rate. not clear if you mean order 10^4 samples at 16384Hz or at (say) 2048Hz.  so, probably better to phrase it as 'a length corresponding to [at least a few seconds] if this is what you mean'

File: pycbc/filter/resample.py [hunk 5]
```diff
@@ -178,6 +178,34 @@ def resample_to_delta_t(timeseries, delta_t, method='butterworth'):
...
+    required to create significant downweigting.
```
Comment (tdent) [inline pycbc/filter/resample.py]: -> 'to produce significant suppression in the notched band'

**Comment by ahnitz** [comment]
@tdent Poke

File: pycbc/filter/resample.py (hunk 2 above)
Comment (tdent) [reply pycbc/filter/resample.py]: The definition as 'number of corrupted samples' is technically correct but it really means just the (one-sided) length of the filter in samples, yes?  So prefer `Extent of the filter on either side of zero`.  Then I would suggest `To achieve frequency resolution df at sampling frequency fs, order should be at least fs/df`

**Comment by ahnitz** [comment]
@tdent Ok, updated text,

**Pull request merged**